### PR TITLE
frontend tests added

### PR DESCRIPTION
**Pull request by suryansh-max**

### CODE REVIEW

File: tests/frontend/test_BoardGUI.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+    def tearDown(self):
+        # Auto cleanup done by unittest
+        pass
+
+    def test_initialization(self):
+        self.assertEqual(self.board_gui.cell_size, self.win_size // self.board_size)
+        self.assertIsInstance(self.board_gui.backend_game, GameManager)
+        self.assertEqual(self.board_gui.count, 0)
+
```
Comment (EXC3ll3NTrhyTHM) [inline tests/frontend/test_BoardGUI.py:30]: Since we also initialize the board_image, black_piece_image and white_piece_image we should write some assertions for those here to reach higher code coverage.

File: tests/frontend/test_BoardGUI.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        mock_smoothscale.return_value = mock_image
+
+        # Call the method with a mock path and resolution and assert the calls were made correctly
+        path = 'resources/board/board.png'
+        resolution = [100, 100]
+        image = self.board_gui.get_scaled_image(path, resolution)
+
+        mock_load.assert_called_once_with(path)
+        mock_smoothscale.assert_called_once_with(mock_image.convert_alpha(), resolution)
+        self.assertEqual(image, mock_image)
```
Comment (EXC3ll3NTrhyTHM) [inline tests/frontend/test_BoardGUI.py:45]: The true test for `get_scaled_image` should be that the returned image is the size of the resolution provided to the function.

Comment (EXC3ll3NTrhyTHM) [reply tests/frontend/test_BoardGUI.py:45]: We would also want a test for a negative path. If image.get_size() is not equal to resoultion then scaled_image.get_size() doesnt equal image.get_size()

Comment (EXC3ll3NTrhyTHM) [reply tests/frontend/test_BoardGUI.py:45]: Also Im not sure if you need to mock out the image and function here. Since we have the actual image path we can use and we can use the `pygame.transform.smoothscale` I think we can just use the actual create the test without any mocks.

File: tests/frontend/test_BoardGUI.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+    @patch('pygame.mouse.get_pos', return_value=(100, 100))
+    @patch('pygame.mouse.get_pressed', return_value=(1, 0, 0))  
+    def test_get_cell_clicked(self, mock_get_pressed, mock_get_pos):
+
+        cell = self.board_gui.get_cell_clicked()
+        expected_cell = pg.Vector2(100, 100) // self.board_gui.cell_size
+        self.assertEqual(cell, expected_cell)
+        mock_get_pos.assert_called_once()
+        mock_get_pressed.assert_called_once()
```
Comment (EXC3ll3NTrhyTHM) [inline tests/frontend/test_BoardGUI.py:55]: Could have an additional test for negative path. `pg.mouse.get_pressed()[1]` - right click then no cell is returned or no click(hover) then no cell is returned.

Comment (EXC3ll3NTrhyTHM) [reply tests/frontend/test_BoardGUI.py:55]: Additional test if all_pieces_placed is true.

Comment (EXC3ll3NTrhyTHM) [reply tests/frontend/test_BoardGUI.py:55]: Currently `get_cell_clicked` also places the piece and changes the turn. May add those tests here if we dont decide to move those steps out of this function.

File: tests/frontend/test_BoardGUI.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        
+        pg.image.load = Mock()
+        pg.transform.smoothscale = Mock(return_value=Mock()) 
+       
+        self.board_size = 8  
+        self.win_size = 640  
+        self.total_pieces = 16
+        self.board_gui = BoardGUI(self.mock_game, self.win_size, self.board_size, self.total_pieces)
+
+    def tearDown(self):
```
Comment (EXC3ll3NTrhyTHM) [inline tests/frontend/test_BoardGUI.py:22]: We can remove the tearDown function since it doesnt do anything.

File: tests/frontend/test_BoardGUI.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    def test_draw_piece(self):
+ 
+        position = (100, 100)
+        self.board_gui.draw_piece('black', position)
+        self.mock_game.screen.blit.assert_called_once_with(self.board_gui.black_piece_image, position)
+
+        self.mock_game.screen.blit.reset_mock()  
+        self.board_gui.draw_piece('white', position)
+        self.mock_game.screen.blit.assert_called_once_with(self.board_gui.white_piece_image, position)
+
```
Comment (EXC3ll3NTrhyTHM) [inline tests/frontend/test_BoardGUI.py:66]: Add assert that the element exist on the screen at the coordinate after `draw_piece` is called.

Comment (EXC3ll3NTrhyTHM) [reply tests/frontend/test_BoardGUI.py:66]: Also can test that if invalid position is given then error is thrown.

File: tests/frontend/test_BoardGUI.py [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    def test_draw_board(self):
+        self.board_gui.draw_board()
+        self.mock_game.screen.blit.assert_called_once_with(self.board_gui.board_image, (0, 0))
+
```
Comment (EXC3ll3NTrhyTHM) [inline tests/frontend/test_BoardGUI.py:70]: Assert that element exists on screen.
